### PR TITLE
David rh patch 1

### PR DESCRIPTION
**Pull request by david-rh**
Adding script to install keycloak-operator to each users namespace

### CODE REVIEW

**Comment by theckang** [comment]
My suggestion: Let's pull the YAML out the script as an individual file.  It'll be easier to update it when we get a certified keycloak, and it'll be consistent with the rest of the instructions (so move the for loop into the README directly).

Tangentially, I'll change the RBACs into a separate file too.  I left it out in haste, but not good practice I think.

File: deployment/workshop/README.md [hunk 1]
```diff
@@ -192,10 +192,13 @@ We have each student install Keycloak as part of a security lab but you need to
...
-* The details will slide in from the right, Click Install (we are using version 8.0.2 today, higher should be OK too)
-* Install for every user project 
-  * eek, that's a lot! I know there's a CLI for this - *TODO replace GUI with CLI instructions*
-  * we plan to automate all this so don't worry it'll be easier soon
+* The details will slide in from the right. Note the version and see that it matches the version in the
+* file keycloak/keycloak-operator-install.sh
+*
+* Install for every user project using the following script
+```
+   keycloak/keycloak-operator-install.sh
```
Comment (theckang) [inline deployment/workshop/README.md:200]: The keycloak install script is not executable by default.  Let's remove the script and use `oc create` directly here.

File: deployment/workshop/keycloak/keycloak-operator-install.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+for (( i=1 ; i <= $NUM_USERS ; i++ ))
```
Comment (theckang) [inline deployment/workshop/keycloak/keycloak-operator-install.sh:2]: NUM_USERS is not set in the script so it gives me `(error token is " ")`.

File: deployment/workshop/keycloak/keycloak-operator-install.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  oc apply -f -<<EOF
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: keycloak-operator
+  namespace: user$i
+spec:
+  channel: alpha
+  Install Plan Approval: Automatic
```
Comment (theckang) [inline deployment/workshop/keycloak/keycloak-operator-install.sh:23]: Does the spec accept this key?

Comment (theckang) [reply deployment/workshop/keycloak/keycloak-operator-install.sh:23]: nevermind, it does!  wow.

File: deployment/workshop/keycloak/keycloak-operator-install.sh (hunk 2 above)
Comment (david-rh) [reply deployment/workshop/keycloak/keycloak-operator-install.sh:2]: Sorry, the intent was that would have already been set by the previous step for creating the users and namespaces.

File: deployment/workshop/README.md (hunk 1 above)
Comment (david-rh) [reply deployment/workshop/README.md:200]: or I could change it to
```
  sh keycloak/keycloak-operator-install.sh

Comment (theckang) [reply deployment/workshop/README.md:200]: You could just change the default rwx mode on the file.  Still, I like the idea of being consistent so we don't have the keycloak part using a script but the rest not.

**Comment by david-rh** [comment]
> My suggestion: Let's pull the YAML out the script as an individual file. It'll be easier to update it when we get a certified keycloak, and it'll be consistent with the rest of the instructions (so move the for loop into the README directly).
> 
> Tangentially, I'll change the RBACs into a separate file too. I left it out in haste, but not good practice I think.

If we move it to another file, we will need to preparse the script to substitute the values. I was thinking about using ansible, and letting ansible take care of the template parsing. If your interested, I may make an attempt to convert this over to ansible after the workshop this week.

File: deployment/workshop/keycloak/keycloak-operator-install.sh (hunk 2 above)
Comment (theckang) [reply deployment/workshop/keycloak/keycloak-operator-install.sh:2]: I think we would need to add `export` to the NUM_USERS command to let the script pull this value?  See my comment above about switching to this a YAML that we use `oc create` with.  I should have done that for the RBAC files too.

**Comment by theckang** [comment]
> > My suggestion: Let's pull the YAML out the script as an individual file. It'll be easier to update it when we get a certified keycloak, and it'll be consistent with the rest of the instructions (so move the for loop into the README directly).
> > Tangentially, I'll change the RBACs into a separate file too. I left it out in haste, but not good practice I think.
> 
> If we move it to another file, we will need to preparse the script to substitute the values. I was thinking about using ansible, and letting ansible take care of the template parsing. If your interested, I may make an attempt to convert this over to ansible after the workshop this week.

OH good point.  Templating is such a pain.  Let's just roll with the script for now then.

File: deployment/workshop/README.md (hunk 1 above)
Comment (theckang) [reply deployment/workshop/README.md:200]: I'll accept the PR and add the `sh`.  Thanks!

**Pull request merged**